pandas/tests/test_multilevel.py

```diff
@@ -290,6 +290,22 @@ def test_datetime_object_multiindex(self):
 
         tm.assert_frame_equal(result, expected)
 
+    def test_multiindex_with_na(self):
+        df = DataFrame(
+            [
+                ["A", np.nan, 1.23, 4.56],
+                ["A", "G", 1.23, 4.56],
+                ["A", "D", 9.87, 10.54],
+            ],
+            columns=["pivot_0", "pivot_1", "col_1", "col_2"],
+        )
+        df.set_index(["pivot_0", "pivot_1"], inplace=True)
+
+        df.at[("A", "F"), "col_2"] = 0.0
+
+        assert df.loc[("A", "F")]["col_2"] == 0.0
```

Comment:
Instead of these `assert`s, could you construct `expected = DataFrame(...)` and then use `tm.assert_frame_equal(df, expected)`

Reply:
Yes. I understand that with your suggestion it would be easier to understand the expected output.